ExecuteJobs.java

```diff
@@ -0,0 +1,80 @@
+import java.util.*;
+import java.io.*;
+
+public class ExecuteJobs {
+  public long start(String process) throws IOException {
+    // TODO some error handling
+    Process p = Runtime.getRuntime().exec(process);
+    long pid = p.pid();
+    return pid;
+  }
+
+  public int stop(long pid) throws IOException {
+    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    String s = pInput.readLine();
+    if(s == null) {
+      // pid not found
+      return 2;
```

Comment:
👍 for using error codes, could you use constants as well? In addition to that it would be helpful to return a more structured error type, e.g. the one that contains error code and the message for the API callers to get more insight into what's happened. In general, the more useful error information you can come up with for the callers, the better for this task (and in general).

Reply:
I'll think about this. Are you suggesting that I change the return type from int to some error data structure that contains the code and an error message?

Reply:
yes